src/chess/board.cc

```diff
@@ -450,46 +447,42 @@ MoveList ChessBoard::GeneratePseudolegalMoves() const {
         result.emplace_back(source, destination);
       }
       // Castlings.
-      if (castlings_.we_can_00()) {
-        bool can_castle = true;
-        for (int i = 5; i < 7; ++i) {
-          if (our_pieces_.get(i) || their_pieces_.get(i)) {
-            can_castle = false;
-            break;
-          }
+      auto walk_free = [this](int from, int to, int rook, int king) {
+        for (int i = from; i <= to; ++i) {
+          if (i == rook || i == king) continue;
+          if (our_pieces_.get(i) || their_pieces_.get(i)) return false;
         }
-        if (can_castle) {
-          for (auto x : k00Attackers) {
-            if (IsUnderAttack(x)) {
-              can_castle = false;
-              break;
-            }
-          }
-        }
-        if (can_castle) {
-          result.emplace_back(source, BoardSquare(0, 6));
-          result.back().SetCastling();
+        return true;
+      };
+      // @From may be less or greater than @to. @To is not included in check.
+      auto range_attacked = [this](int from, int to) {
+        const int increment = from < to ? 1 : -1;
+        while (true) {
+          if (IsUnderAttack(from)) return true;
+          if (from == to) break;
+          from += increment;
```

Comment:
Despite the comment, `to` is included in the check to cover the case `from == to`. Suggested alternative (saves a call to `IsUnderAttack()`):
```
      // @From may be less or greater than @to. @To is not included in check
      // unless it is the same with @from.
      auto range_attacked = [this](int from, int to) {
        if (from == to) return IsUnderAttack(from);
        const int increment = from < to ? 1 : -1;
        while (from != to) {
          if (IsUnderAttack(from)) return true;
          from += increment;
        }
        return false;
      };
```